mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -349,14 +361,62 @@ class AffineForLowering : public OpRewritePattern<AffineForOp> {
     Value lowerBound = lowerAffineLowerBound(op, rewriter);
     Value upperBound = lowerAffineUpperBound(op, rewriter);
     Value step = rewriter.create<ConstantIndexOp>(loc, op.getStep());
-    auto f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step);
+    SmallVector<Value, 4> iterArgs(op.getIterOperands().begin(),
+                                   op.getIterOperands().end());
+    scf::ForOp f;
+    if (iterArgs.size() > 0) {
+      f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step,
+                                      iterArgs);
+    } else {
+      f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step);
+    }
     rewriter.eraseBlock(f.getBody());
     rewriter.inlineRegionBefore(op.region(), f.region(), f.region().end());
-    rewriter.eraseOp(op);
+    rewriter.replaceOp(op, f.results());
     return success();
   }
 };
 
+/// This function returns the value of identity value associated with a
+/// AtomicRMWKind op.
+Value getIdentityValue(llvm::Optional<mlir::AtomicRMWKind> op,
+                       OpBuilder &builder, Location loc) {
+  switch (op.getValue()) {
+  case AtomicRMWKind::addf:
+    return builder.create<ConstantOp>(loc, builder.getF32FloatAttr(0));
+  case AtomicRMWKind::addi:
+    return builder.create<ConstantOp>(loc, builder.getI32IntegerAttr(0));
+  case AtomicRMWKind::mulf:
+    return builder.create<ConstantOp>(loc, builder.getF32FloatAttr(1));
+  case AtomicRMWKind::muli:
+    return builder.create<ConstantOp>(loc, builder.getI32IntegerAttr(1));
+  // TODO: Add remaining reduction operations
+  default:
+    emitOptionalError(loc, "Reduction operation type not supported");
+  }
```

Comment:
If these are the only ones currently, use llvm_unreachable at the end and remove the 'default' case.

Reply:
Actually, there are some ``AtomicRMWKind`` which are not handled like max/min.